src/Install.php

```diff
@@ -487,6 +490,57 @@ protected static function create_notes() {
 		WC_Admin_Notes_Historical_Data::possibly_add_note();
 	}
 
+	/**
+	 * Update selected notes, if they need changing from when they were
+	 * originally created.
+	 */
+	protected static function update_notes() {
+		$note_factories_to_update = array(
+			new WC_Admin_Notes_Deactivate_Plugin(),
+		);
+
+		/**
+		 * Filter allowing additional note factories to be updated.
+		 *
+		 * @param array $factories The additional factories.
+		 */
+		$additional_note_factories_to_update = apply_filters(
+			'woocommerce_admin_update_note_factories',
+			array()
+		);
+
+		if ( is_array( $additional_note_factories_to_update ) ) {
+			$note_factories_to_update = array_merge(
+				$note_factories_to_update,
+				$additional_note_factories_to_update
+			);
+		}
+
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		foreach ( $note_factories_to_update as $note_factory ) {
+			$note_ids = $data_store->get_notes_with_name( $note_factory::NOTE_NAME );
+			$new_note = $note_factory->get_note();
+
+			if ( ! $new_note instanceof WC_Admin_Note ) {
+				continue;
+			}
+
+			foreach ( (array) $note_ids as $note_id ) {
+				$new_note->set_id( $note_id );
+
+				// Copy some state-based values from the database.
```

Comment:
Why only same state-based values? See above for general confusion.

Reply:
I've updated the comment here to explain things a little better (see above).